molvis/molvisgl/src/processors/molecularrenderer.cpp

```diff
@@ -0,0 +1,194 @@
+                       {"ballAndStick", "Ball & Stick", Representation::BallAndStick}},
+                      0)
+    , radiusScaling_("radiusScaling", "Radius Scaling", 1.0f, 0.0f, 2.0f)
+    , forceRadius_("forceRadius", "Force Radius", false, InvalidationLevel::InvalidResources)
+    , defaultRadius_("defaultRadius", "Default Radius", 0.15f, 0.00001f, 2.0f, 0.01f)
+    , camera_("camera", "Camera", util::boundingBox(inport_))
+    , lighting_("lighting", "Lighting", &camera_)
+    , trackball_(&camera_)
+
+    , vdwShaders_{{{ShaderType::Vertex, std::string{"vdw.vert"}},
+                   {ShaderType::Geometry, std::string{"vdw.geom"}},
+                   {ShaderType::Fragment, std::string{"vdw.frag"}}},
+                  {{BufferType::PositionAttrib, MeshShaderCache::Mandatory, "vec3"},
+                   {BufferType::ColorAttrib, MeshShaderCache::Optional, "vec4"},
+                   {BufferType::RadiiAttrib, MeshShaderCache::Optional, "float"},
+                   {BufferType::PickingAttrib, MeshShaderCache::Optional, "uint"},
+                   {BufferType::ScalarMetaAttrib, MeshShaderCache::Optional, "float"}},
+
+                  [&](Shader& shader) -> void {
+                      shader.onReload(
+                          [this]() { invalidate(InvalidationLevel::InvalidResources); });
+                      configureVdWShader(shader);
+                  }}
+    , licoriceShaders_{
+          {{ShaderType::Vertex, std::string{"licorice.vert"}},
+           {ShaderType::Geometry, std::string{"licorice.geom"}},
+           {ShaderType::Fragment, std::string{"licorice.frag"}}},
+          {{BufferType::PositionAttrib, MeshShaderCache::Mandatory, "vec3"},
+           {BufferType::ColorAttrib, MeshShaderCache::Optional, "vec4"},
+           {BufferType::ScalarMetaAttrib, MeshShaderCache::Optional, "float"}},
+
+          [&](Shader& shader) -> void {
+              shader.onReload([this]() { invalidate(InvalidationLevel::InvalidResources); });
+              configureLicoriceShader(shader);
+          }} {
+    addPort(inport_);
+    addPort(imageInport_);
+    addPort(outport_);
+
+    imageInport_.setOptional(true);
+
+    addProperties(representation_, radiusScaling_, forceRadius_, defaultRadius_, camera_, lighting_,
+                  trackball_);
+}
+
+void MolecularRenderer::process() {
+    utilgl::activateTargetAndClearOrCopySource(outport_, imageInport_);
+
+    auto drawMesh = [](std::shared_ptr<const Mesh> mesh, MeshDrawerGL::DrawObject& drawer,
+                       DrawType dt) {
+        if ((mesh->getNumberOfIndicies() == 0) && (mesh->getDefaultMeshInfo().dt == dt)) {
+            drawer.draw();
+        } else {
+            for (size_t i = 0; i < mesh->getNumberOfIndicies(); ++i) {
+                if (mesh->getIndexMeshInfo(i).dt == dt) {
+                    drawer.draw(i);
+                }
+            }
+        }
+    };
+
+    auto drawVdW = [&](float radius) {
```

Comment:
What is this VdW? :)
Why not spell it out?

Reply:
see comment above. VDW is a standard abbreviation in molvis.